src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -299,13 +299,20 @@ private class TimerState
 
         private static void HandleHeartbeat(object state)
         {
-            var timerState = (TimerState)state;
-            if (timerState.Parent.TryGetTarget(out BulkInsertOperation bulkInsert) == false)
+            try
             {
-                timerState.Timer.Dispose();
-                return;
+                var timerState = (TimerState)state;
+                if (timerState.Parent.TryGetTarget(out BulkInsertOperation bulkInsert) == false)
+                {
+                    timerState.Timer.Dispose();
+                    return;
+                }
+                _ = bulkInsert.SendHeartBeatAsync();
```

Comment:
you need to add try/catch inside the SendHeartBeatAsync as well, this is a background task, not awaited so wrapping it in try/catch does not yield any results

Reply:
Done